src/otx/algorithms/common/adapters/mmcv/configurer.py

```diff
@@ -264,18 +264,11 @@ def configure_fp16(cfg: Config):
                 opts["type"] = "HPUOptimizerHook"
                 cfg.optimizer_config.update(opts)
             elif is_xpu_available():
-                opts.update({"distributed": distributed, **fp16_config})
                 if optim_type == "SAMOptimizerHook":
                     logger.warning("SAMOptimizerHook is not supported on XPU yet, changed to OptimizerHook.")
                     opts["type"] = "OptimizerHook"
-                if optim_type == "OptimizerHook":
-                    opts["type"] = "BFp16XPUOptimizerHook"
```

Comment:
I think removing `cfg.fp16_` is required as well because of setting `enable_autocast` to `False`.

For example,
https://github.com/openvinotoolkit/training_extensions/blob/c1beb0549e58f18c3e9f17097d2a02e52bd81231/src/otx/algorithms/detection/adapters/mmdet/apis/train.py#L121-L123

Reply:
I think `torch.autocast` in `XPUDataParallel` should be removed. When I checked `custom_auto_fp16` code it already uses `torch.autocast`. So, it's duplicated. After removing it, we don't have to remove `cfg.fp16_` because `enable_autocast` disappear after update.

Reply:
done at [@48f463a](https://github.com/openvinotoolkit/training_extensions/pull/2683/commits/48f463ac9682ce6a0f550420d33c0201d4f52291)